Component/button

PReichetanz opened the pull request:
Create main button as controlled component

- Add styling according to design
- Add general handleClick function

<img width="458" alt="Bildschirmfoto 2021-11-25 um 12 24 50" src="https://user-images.githubusercontent.com/33567019/143434013-ce46ff49-5d6d-4ad9-bb08-52e033b382a0.png">

[View in storybook](https://capstone-rev-component--2uz5hj.herokuapp.com/storybook/?path=/story/component-button--regular)

PReichetanz committed:
Refactor existing button to addButton

PReichetanz committed:
Merge branch 'main' into component/button

PReichetanz committed:
Remove button.module.css

PReichetanz committed:
Rename Button to MainButton

marwinburesch commented on src/app/components/MainButton/MainButton.tsx:
```diff
@@ -0,0 +1,25 @@
+  children: ReactNode;
+  handleClick: () => void;
+};
+export default function MainButton({
+  children,
+  handleClick,
+}: ButtonProps): JSX.Element {
+  return <Button onClick={handleClick}>{children}</Button>;
+}
```
This doesn't need to be another component returning a styled component, you could just export the styled component and pass it your ButtonProps type :)

marwinburesch reviewed (commented):
Almost as awesome as it can be!, you're 99% there :)

PReichetanz committed:
Refactor MainButton after review

- Remove function MainButton
- Export default styled component "MainButton"
- Add ButtonProps to MainButton styled component
